### ISSUE TITLE
Option for NHWC convolution

### ISSUE THREAD

**Issue by timmoon10** [LLNL/lbann#698]
Tensor cores on Volta GPUs expect data in NHWC format and converting NCHW to NHWC can take a significant fraction of runtime (https://devblogs.nvidia.com/tensor-core-ai-performance-milestones/). This is not a concern for us right now since we don't yet support half precision (and it seems NCHW format is faster with non-tensor-core cuDNN, see https://github.com/tensorflow/tensorflow/issues/8286), but it's something to keep on our radar. If we provide the option for NHWC convolutions, we'll need to make appropriate changes to pooling, batch norm, data readers, and anything other components that expect image data.
